### ISSUE TITLE
questions about output pins

### ISSUE THREAD

**Issue by leobrowning92** [enveloop/miniMO#3]
Is there a reason that you have made 2 output headers?

also, do you have plans for/ have you considered a design using mono jacks instead of pin headers? On the one hand it might make it more accessible as people can use their existing mono/aux to interface but i guess it might drive price up.

Cheers

**Comment by enveloop** [comment]
//Is there a reason that you have made 2 output headers?

Yes, several: (for the sake of clarification, we're talking about headers 1 and 2)
*I can check a module's output without disconnecting it from whatever it is modulating; 
*I can modulate two things at once (e.g., ADSR controlling both OSC and Filter)
*I like the symmetry with the I/O 3 and 4;
*A free header within this pair doubles as a mixing input. If you are using I/O 1 to listen to module A, you can connect module B's I/O 1 (or 2) to modules A's I/O 2 and also listen to B through A's I/O 1. In fact, you can daisy chain several modules that way. I didn't realize I could do this until recently so I think of it as a hack, but it's actually quite useful :D

// have you considered a design using mono jacks instead of pin headers? 
 
Again, yes -I even built several prototypes using jacks. It is as you say, jacks are more common and everyone has a couple around at home, but the inputs add a lot to a module's cost (compared to the other components); also, jack cables are much more expensive than header wires unless you make them yourself, in which case you might as well use headers and make extra adapters as needed.

Thanks for your questions!

**Comment by leobrowning92** [comment]
the daisy chaining sounds fantastic! I would love to get some more detail as to how it actually can be used for something interesting/useful, as it seems to me that this would only be possible if the output from 1  wouldn't affect the input from 2. Do you have code on the repo that uses them as a mixed input?Id like to have a look.

also small question. it seems like female headers and male header wires is the most common for patch bays etc. was the decision to use male headers again a cost consideration?

Cheers

**Comment by enveloop** [comment]
//this would only be possible if the output from 1 wouldn't affect the input from 2
It doesn't, as far as I can tell. Off the top of my head, it's a simple way to combine several oscillators in a single output -see [here.](https://twitter.com/miniMOsynth/status/919383247527862272)

//Do you have code on the repo that uses them as a mixed input?
There's no code, this feature is analog xD

//was the decision to use male headers again a cost consideration?

I think male headers are cheaper, yes. Also, I connect the modules with pairs of wires to ensure they always share grounds, and the female wires I have around are easy to glue together and fit securely inside the cases I designed; male wires often have cylindrical ends, so they're hard to manipulate in pairs, and in any case they stick out quite a bit. I built one or two modules with female headers for tests, but I think male headers are a better choice for this particular design. 

That said, having the exposed male pins allows you to tap on the inputs with your fingers for some interesting effects, and lately I'm exploiting this "hack" by using I/O4 as a second button  (in [LunaMod](https://github.com/enveloop/miniMO/blob/master/Programs/LunaMod_Generator/LunaMod_Generator.ino) and the [Low Power Beacon](https://github.com/enveloop/miniMO/blob/master/Programs/Low_Power_Beacon/Low_Power_Beacon.ino), at the moment).

**Comment by leobrowning92** [comment]
Cool, so perhaps calling them mixed inputs isn't the best way to phrase it, as the attiny cant actually process an input signal through them. Instead they facilitate daisychaining of outputs. 

Im getting more and more excited to build some of these so I can get playing! the parts are still in the mail, but ive just designed the PCB to encorporate them. was there a reason you went with 38x38mm square instead of an even 40x40mm?

Cheers again for being so cool about all of my questions!

**Comment by enveloop** [comment]
//Cool, so perhaps calling them mixed inputs isn't the best way to phrase it, as the attiny cant actually process an input signal through them. 

In the very first version of the board I named 1&2 "OUT", and 3&4 "IN 1-2"; later on I realized that I could use input 2 as an output, so I decided to rename them to I/O 1-4 (as in, a group including inputs and outputs), and specify what they actually do in the manual at the beginning of every program. 

In the case of I/O 1&2, they could be used as mixed inputs (PB4 as ADC2), though I didn't describe them as such (I wrote that a free header can act as a mixing input, adding to the other's output). I don't think I'll ever use them as "processing" inputs, but if I did, but the naming convention already accounts for that.

//was there a reason you went with 38x38mm square instead of an even 40x40mm?

It's the smallest I could design it :P 
Conveniently, if you add an enclosure surrounding the board with walls 1mm thick, you reach 40x40. The female "programming" headers protrude from the board, and that's so that they are flush with the enclosure.   
 
//Cheers again for being so cool about all of my questions!
Thanks for asking! I'm kind of wishing I had a forum, rather than having them under issues, though.
